chalk-recursive/src/solve.rs

```diff
@@ -154,11 +154,6 @@ trait SolveIterationHelpers<I: Interner>: SolveDatabase<I> {
         for program_clause in clauses {
             debug_span!("solve_from_clauses", clause = ?program_clause);
 
-            // If we have a completely ambiguous answer, it's not going to get better, so stop
```

Comment:
we could actually early-exit here if `cur_solution` `is_trivial_and_always_true` though, right? that might actually save us quite a bit of time, depending on how the clauses are ordered.

Reply:
I did this below